ClemBot.Api/ClemBot.Api.Core/Features/Tags/Details.cs

```diff
@@ -37,10 +37,19 @@ public record QueryHandler(ClemBotContext _context) : IRequestHandler<Query, IQu
     {
         public async Task<IQueryResult<Model>> Handle(Query request, CancellationToken cancellationToken)
         {
+            // var tag = await _context.Tags
+            //     .Where(g => g.GuildId == request.GuildId && g.Name == request.Name)
+            //     .Include(y => y.TagUses)
+            //     .FirstOrDefaultAsync();
+
             var tag = await _context.Tags
-                .Where(g => g.GuildId == request.GuildId && g.Name == request.Name)
-                .Include(y => y.TagUses)
-                .FirstOrDefaultAsync();
+                .Where(t => t.GuildId == request.GuildId)
+                .Select(t => new {t, Similarity = EF.Functions.TrigramsSimilarity(t.Name, request.Name)})
+                .Where(t => t.Similarity > 0.4)
+                .OrderByDescending(t => t.Similarity)
+                .Select(t => t.t)
+                .Include(t => t.TagUses)
+                .FirstOrDefaultAsync(cancellationToken);
```

Comment:
we dont want to run the trigram search on a details call, only on an actual search

Reply:
This endpoint is the one called in the tags service for each regex match. Should I make a new endpoint or change the invoke endpoint?

Reply:
We dont want to fuzzy match on inline syntax, if the tag doesnt exist we assume the user wasnt attempting to invoke a tag